test/unit/luaTable.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { TSTLErrors } from "../../src/TSTLErrors";
+import * as util from "../util";
+
+const tableLib = `
+/** @LuaTable */
+declare class Table<K extends {} = {}, V = any> {
+    length: number;
+    constructor(notAllowed?: boolean);
+    set(key?: K, value?: V): void;
+    get(key?: K): V;
+    other(): void;
+}
+declare let tbl: Table;
+`;
```

Comment:
It would be good to run these tests against a `@luaTable` interface as well. Perhaps have 2 lib strings and do a `test.each()` to try both?